tests/sentry/debug_files/test_artifact_bundle_indexing.py

```diff
@@ -605,3 +615,75 @@ def test_flat_file_index_with_index_stored_and_duplicated_bundle(self):
             "files_by_url": {"~/path/to/app.js": [0]},
             "files_by_debug_id": {},
         }
+
+    def test_flat_file_index_enforces_limits(self):
+        # The first "bundle limit" test needs 2 minutes to run, the complete test
+        # does not finish at all in reasonable time.
+        # I'm just losing my mind how python / pytest can be *this* slow?
+        return
```

Comment:
I mean really… what the hell is going on here?

Reply:
Is it because of the size of the input? Are you sure the code is not making any db calls as side effect?

Reply:
The tests only touching `FlatFileIndex` are not making any db calls.

I expect these things really are slow because of the size of the input.
`_index_and_bundle_meta_for_id` for one is doing a linear scan for the bundle.
And then actual removal iterates over every url / debug-id within the bundle, which is not cheap either. But the slowness is still unreasonable. Maybe if it is indeed expensive enough, might be worth rewriting it in Rust :-)